src/language-server/diagnostics.ts

```diff
@@ -0,0 +1,241 @@
+/**
```

Comment:
is this mainly the same code as before but just reorganized into this file? skimmed through this file and looks like it, but didn't want to miss any functionality changes, if any